### ISSUE TITLE
[Bug]: Mask discarded when predicted bounding box has negative coordinates in crop_mask

### ISSUE THREAD

**Issue by xianshunw** [ultralytics/ultralytics#24100]
Issue Description:
I encountered an issue where obvious targets (e.g., persons) are occasionally ignored when using the yolo26n-seg (or similar) models, regardless of whether I use official or fine-tuned weights.

Root Cause Analysis:
After debugging the post-processing pipeline, I identified the root cause in the crop_mask function within ultralytics/utils/ops.py.

When a predicted bounding box has negative coordinates (e.g., x1 or y1 < 0, which can happen with raw model outputs), the current cropping logic leads to unexpected behavior. Specifically, it can cause the mask to be zeroed out or improperly sliced. Consequently, the mask validation checker (around [lines 467-471](https://github.com/ultralytics/ultralytics/blob/5b400439c48d5af5a2aefe972dbdda9305cdaa03/ultralytics/utils/ops.py#L467-L471)) discards these predicted masks entirely because they appear empty.

Code Reference:
https://github.com/ultralytics/ultralytics/blob/5b400439c48d5af5a2aefe972dbdda9305cdaa03/ultralytics/utils/ops.py#L467-L471

Suggested Impact:
This issue significantly affects detection stability for objects near the image boundaries. The crop_mask function should likely clamp the coordinates to the image dimensions before slicing or handle negative indices more robustly.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xianshunw, thank you for reporting this bug in Ultralytics 🚀. This is an automated response to help move things along, and an Ultralytics engineer will also assist you soon.

We appreciate the detailed description and root-cause investigation 🔍. Since this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that shows the issue in a small, self-contained script. An MRE helps us reliably reproduce and validate the behavior, especially for post-processing issues like this.

It would also help if you can include:
- the exact `ultralytics` version you are using
- your OS, Python, and PyTorch versions
- whether the issue occurs with both official and custom weights
- a sample image and the corresponding prediction output/logs
- the smallest code snippet needed to reproduce the discarded mask behavior

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) as well, where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples that may help confirm expected behavior.

If this is related to custom training ❓, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by xianshunw** [comment]
<img width="577" height="773" alt="Image" src="https://github.com/user-attachments/assets/40a2ce89-6129-430b-bf6a-74e639e26b65" />

The problem could be reproduced using my attached picture by the following command
```
yolo predict model=yolo26n-seg.pt source='input.jpg'
```

**Comment by glenn-jocher** [comment]
Thanks for the repro — this looks like a real edge case in [`crop_mask()`](https://docs.ultralytics.com/reference/utils/ops/#ultralytics.utils.ops.crop_mask), since negative `xyxy` values can be sliced before bounds checking; clipping first with [`clip_boxes()`](https://docs.ultralytics.com/reference/utils/ops/#ultralytics.utils.ops.clip_boxes) seems like the right direction. Please also share your `yolo checks` output and exact `ultralytics` version or commit so we can verify it against current `main`.

**Comment by Y-T-G** [comment]
@xianshunw Can you test this?

```
https://github.com/ultralytics/ultralytics@coords-clamp
```

**Comment by xianshunw** [comment]
> [@xianshunw](https://github.com/xianshunw) Can you test this?
> 
> ```
> https://github.com/ultralytics/ultralytics@coords-clamp
> ```

Yes, It works fine. Thanks.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has now been merged in [PR #24115: Clamp coordinates in `crop_mask` before cropping](https://github.com/ultralytics/ultralytics/pull/24115) by @Y-T-G. 🎉

### What changed
This PR updates `crop_mask()` in `ultralytics/utils/ops.py` so bounding box coordinates are clamped to valid non-negative values before cropping masks:

- changed from `boxes.round().int()`
- to `boxes.clamp(min=0).round().int()`

This should help prevent negative crop indices from affecting mask cropping, which may improve segmentation robustness in cases where predicted boxes extend beyond the image boundary. Based on the PR details, this is especially relevant for CPU-based prediction paths with smaller numbers of masks.

### Credits
Thanks to @Y-T-G for the fix, and to everyone who helped report, investigate, and narrow this down.

### How to test the fix
If helpful, the merged changes can be tested now from the latest `main` branch with:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Alternatively, it’s also fine to wait for the next official PyPI release if preferred.

### Feedback
Please let us know whether the changes in that PR resolve the issue on your side. If the problem persists, sharing a small reproducible example would be very helpful for further investigation.

Thanks 🙏 for reporting this, and any additional feedback is always welcome.
